src/core/border_router/routing_manager.cpp

```diff
@@ -623,11 +623,18 @@ void RoutingManager::StartRoutingPolicyEvaluationDelay(void)
 
     uint32_t randomDelay;
 
+    // Do not start delay timer if Router Solicitation is in progress.
+    // BR will evaluate Routing Policy after Router Solicitation is complete.
+    VerifyOrExit(!IsRouterSolicitationInProgress());
```

Comment:
I think we shouldn't skip routing policy evaluation during Router Solicitation:
1. we need to update the RA header parameters and resend our RA messages if received a RA which is sent from the infra if;
2. we need to advertise OMR prefix in both Thread & infra network if Thread network data have changes;

We are already skipping on-link prefix evaluation if we are doing RS and external route for discovered on-link prefix will be added immediately when receiving the PIO. That said, bi-directional connectivity will be established before finishing RS.
```c++
    // We don't evaluate on-link prefix if we are doing Router Solicitation.
    VerifyOrExit(!mRouterSolicitTimer.IsRunning(),
                 newOnLinkPrefix = (mIsAdvertisingLocalOnLinkPrefix ? &mLocalOnLinkPrefix : nullptr));
```
https://github.com/openthread/openthread/blob/main/src/core/border_router/routing_manager.cpp#L501-L503

Reply:
make sense to me. 
Closing this PR.